docs/guides/deploy-hyperlane-local-agents.mdx

```diff
@@ -0,0 +1,397 @@
+
+```sh
+...
+-e CONFIG_FILES=/config/agent-config.json \
+--mount type=bind,source=$CONFIG_FILES,target=/config/agent-config.json,readonly \
+--mount type=bind,source="$(pwd)"/hyperlane_db_validator_<your_chain_name>,target=/hyperlane_db \
+--mount type=bind,source="$(pwd)"/$VALIDATOR_SIGNATURES_DIR,target=/tmp/validator-signatures \
+...
+```
+
+Hardcoding these paths deduplucates the configuration, making it easier to pass the right arguments when running the container. See the example below, where that's left is for you to configure the chain name and validator key.
+
+```sh
+...
+./validator \
+--db /hyperlane_db \
+--originChainName <your_chain_name> \
+--checkpointSyncer.type localStorage \
+--checkpointSyncer.path /tmp/validator-signatures \
+--validator.key <your_validator_key>
+...
+```
+
+</TabItem>
+<TabItem value="from-source" label="Building from source">
+
+**Clone and setup**
+
+First, clone the Hyperlane monorepo:
+
+```sh
+git clone git@github.com:hyperlane-xyz/hyperlane-monorepo.git
+```
+
+Then follow the [setup instructions](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/README.md) in the `rust` directory. This should setup `rustup` as well as Rosetta 2 if you are on Apple Silicon.
+
+```sh
+# install rustup
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+
+# (apple silicon only) install rosetta 2
+softwareupdate --install-rosetta --agree-to-license
+```
+
+</TabItem>
+</Tabs>
+
+### Run
+
+<Tabs groupId="docker">
+<TabItem value="docker" label="Using Docker">
+Now that you understand more about configuring validator arguments, pull the latest docker image:
+
+```sh
+docker pull gcr.io/abacus-labs-dev/hyperlane-agent:main
+```
+
+Finally, run the validator:
+
+```sh
+docker run \
+  -it \
+  -e CONFIG_FILES=/config/agent-config.json \
+  --mount type=bind,source=$CONFIG_FILES,target=/config/agent-config.json,readonly \
+  --mount type=bind,source="$(pwd)"/hyperlane_db_validator_<your_chain_name>,target=/hyperlane_db \
+  --mount type=bind,source="$(pwd)"/$VALIDATOR_SIGNATURES_DIR,target=/tmp/validator-signatures \
+  gcr.io/abacus-labs-dev/hyperlane-agent:main \
+  ./validator \
+  --db /hyperlane_db \
+  --originChainName <your_chain_name> \
+  --checkpointSyncer.type localStorage \
+  --checkpointSyncer.path /tmp/validator-signatures \
+  --validator.key <your_validator_key>
+```
+
+</TabItem>
+
+<TabItem value="from-source" label="Building from source">
+
+After following the setup instructions, you should now be able to use `cargo` to run the Validator:
+
+```sh
+cargo run --bin validator -- \
+    --db ./hyperlane_db_validator_<your_chain_name> \
+    --originChainName <your_chain_name> \
+    --checkpointSyncer.type localStorage \
+    --checkpointSyncer.path $VALIDATOR_SIGNATURES_DIR \
+    --validator.key <your_validator_key>
+```
+
+:::note (optional) Run the binary directly
+
+You can alternatively build out the agent:
+
+```sh
+cargo build --release bin validator
```

Comment:
if we're using the release flag here, we should be using it in the `cargo run` command above as well